### DISCUSSION TITLE
git push gets stuck after "total" line

### DISCUSSION THREAD

**Discussion by ospaarmann** [community/community#56937] [Repositories]
### Select Topic Area

Bug

### Body

Sometimes git gets stuck on push after the "Total" line. It happens completely randomly. Sometimes it works again after restarting my MacBook. I have this issue since a couple of weeks and it is really annoying. I think there is everything ok on my side. I'm authenticated and the push size is not really big. Internet connection is also fine.

The complete push was 2.71 MiB. Git gets stuck on GitHub Desktop and on the command line. This happens more and more often and really needs to be fixed. 

```bash
🤷 ❯ git push
Enumerating objects: 30, done.
Counting objects: 100% (30/30), done.
Delta compression using up to 10 threads
Compressing objects: 100% (24/24), done.
Writing objects: 100% (27/27), 2.72 MiB | 491.00 KiB/s, done.
Total 27 (delta 11), reused 0 (delta 0), pack-reused 0
```

```bash
🤷 ❯ git --version
git version 2.38.1
```

I'm on a MacBook Pro, 14-inch, 2021, Apple M1 Max, 64 GB RAM.

**Does anyone have a solution for this?**

**Comment by kiashok** [comment]
running into the same recently! @ospaarmann did you happen to find a resolution?

**Comment by dmrd2653** [comment]
Enter this in the command line: 

`git config --global http.postBuffer 157286400`

**Reply by USNA2014** [reply]
Didn't work :(

**Reply by osmaircoelho** [reply]
It didn't work for me as well.

any tips?

**Reply by yakheda** [reply]
Not working(

**Reply by yunylz** [reply]
not working..

**Reply by devfemibadmus** [reply]
not working

**Reply by skipper-83** [reply]
works for me, thanks!

**Comment by devfemibadmus** [comment]
below solve mine, got stuck

- delete credential and re authenticate
- add new commit
- try pulling
- do `git config --global credential.helper manager`
- now `git push origin <branch> --force`
- if still stuck close current terminal and try in new one
- wait some whiles then u see everything is up todate
- run git push again to confirm

**Reply by baljimaurya** [reply]
Not working

**Comment by somedeveloper00** [comment]
`git push --progress`
->
```
Enumerating objects: 17739, done.
Counting objects: 100% (17739/17739), done.
Delta compression using up to 16 threads
Compressing objects: 100% (7582/7582), done.
Writing objects: 100% (17738/17738), 540.28 MiB | 383.66 MiB/s, done.
Total 17738 (delta 10147), reused 17738 (delta 10147), pack-reused 0 (from 0)
```
stuck there. From Task Manager, `Git for Windows` is consuming network to upload at a very speed (1.5Mb/s whereas normally I could reach at least 5Mb/s according to benchmarks)

**Reply by somedeveloper00** [reply]
+ If it *is* uploading, some progress indicator would be necessary for long running tasks like this

**Reply by somedeveloper00** [reply]
+ This does NOT happen for the same project but with a custom ssh server. It only happens with Github
